### PR TITLE
NCSD-986: App gateway NSG

### PR DESCRIPTION
**Pull request by rivoric**
Creates a NSG for use by the app gateway

### CODE REVIEW

File: Resources/template.json [hunk 1]
```diff
@@ -141,46 +141,72 @@
...
+      "defaultValue": "0.0.0.0/1,128.0.0.0/2,192.0.0.0/3,224.0.0.0/4",
+      "metadata": {
+        "description": "Restricts HTTP(S) traffic to App Gateway to the passed in range"
+      }
+    }
   },
   "variables": {
     "dfcDevOpsTemplatesUrl": "https://raw.githubusercontent.com/SkillsFundingAgency/dfc-devops/master/ArmTemplates/",
-    "localTemplatesUrl": "https://raw.githubusercontent.com/SkillsFundingAgency/dfc-shared/master/Resources/",
+    "localTemplatesUrl": "https://raw.githubusercontent.com/SkillsFundingAgency/dfc-shared/NCSD-986_appgatewaynsg/Resources/",
```
Comment (NickGraham101) [inline Resources/template.json]: still referring to the branch rather than master

File: Resources/template.json [hunk 2]
```diff
@@ -544,6 +570,38 @@
...
+        },
+        "parameters": {
+          "name": {
+            "value": "[parameters('appGatewayNetworkSecurityGroup')]"
+          },
+          "agSubnetName": {
+            "value": "[parameters('appGatewaySubnetName')]"
+          },
+          "vnetName": {
+            "value": "[if(equals(parameters('envType'),'dev'), reference('sharedManagedInstanceEnvironment','2017-05-10').outputs.virtualNetworkName.value, parameters('appGatewayVnetName'))]"
```
Comment (NickGraham101) [inline Resources/template.json]: Should this condition just be testing for 'dev' or 'dev' & 'sit'?

Comment (rivoric) [reply Resources/template.json]: Just dev, the description for the param says "Type of environment, either dev or prod"

File: Resources/template.json (hunk 1 above)
Comment (rivoric) [reply Resources/template.json]: Doh, fixed

**Pull request merged**